tests/unit/versioning/VersionID.spec.js

```diff
@@ -9,83 +10,124 @@ function randkey(length) {
     return key;
 }
 
-describe('test generating versionIds', () => {
-    const count = 1000;
+function generateRandomVIDs(count) {
     const vids = Array(count).fill(null);
     for (let i = 0; i < count; i++) {
         vids[i] = VID.generateVersionId(randkey(15), 'PARIS');
     }
     process.env.VID_CRYPTO_PASSWORD = randkey(64);
+    return vids;
+}
+
+const count = 1000000;
+
+describe('test generating versionIds', () => {
+    describe('legaxy hex encoding', () => {
+        env.S3_VERSION_ID_ENCODING_TYPE = 'hex';
+        const vids = generateRandomVIDs(count);
 
-    it('sorted in reversed chronological and alphabetical order', () => {
-        for (let i = 0; i < count; i++) {
-            if (i !== 0) {
-                assert(vids[i - 1] > vids[i],
-                    'previous VersionID is higher than its next');
+        it('sorted in reversed chronological and alphabetical order', () => {
+            for (let i = 0; i < count; i++) {
+                if (i !== 0) {
```

Comment:
You can just start at 1 to avoid this extra check

Reply:
with branch prediction this is going to be barely noticeable 
https://measurethat.net/Benchmarks/Show/18409/0/branch-prediction
I don't see the need for micro-optimizing unit tests in general, but that's just my opinion.

Reply:
My comment was not about optimization it was about reducing a tab and readability. I rarely try and optimize tests unless its like super obvious or less code in general.